plugins/serializers/json/json.go

```diff
@@ -49,8 +50,26 @@ func (s *serializer) SerializeBatch(metrics []telegraf.Metric) ([]byte, error) {
 
 func (s *serializer) createObject(metric telegraf.Metric) map[string]interface{} {
 	m := make(map[string]interface{}, 4)
-	m["tags"] = metric.Tags()
-	m["fields"] = metric.Fields()
+
+	tags := make(map[string]string, len(metric.TagList()))
+	for _, tag := range metric.TagList() {
+		tags[tag.Key] = tag.Value
+	}
+	m["tags"] = tags
```

Comment:
Just curious why this is better than before.  Is it better not to refer to the same map?  Kind of a deep copy but what's the benefit?

Reply:
I'd like to remove the `metric.Tags()` and `Fields()` method the interface, they are marked as deprecated now, so I just took the liberty to do stop using them here.  We don't use a map in the metric implementation anymore, since a list is more memory and GC friendly in most cases.  Both of these functions make a new map and are basically equivalent to what I've done here.

Reply:
makes sense, thanks